script/version-utils.js

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env node
+
+const { logger, spawnOrFail, prompt, shouldContinuePrompt, quit, fs, path } = require('./cli-utils');
+const currentVersion = require('../package.json').version;
+
+const isPreRelease = (version) => {
+  //Check whether there is a build version at the end (e.g., 3.0.0-beta.0)
+  return version.split('.')[3] >= 0;
+};
+
+const getNewVersion = (currentVersion, versionIncrement) => {
+  const verArr = currentVersion.split('.');
+  const isBeta = isPreRelease(currentVersion);
+  switch (versionIncrement) {
+    case 1: //Patch
+      if (isBeta) {
+        logger.error(`ERROR: Cannot increase patch in pre-release version `);
+        return undefined;
+      }
+      verArr[2] = Number(verArr[2]) + 1;
+      return verArr.join('.');
+    case 2: // Minor
+      if (isBeta) {
+        logger.error(`ERROR: Cannot increase patch in pre-release version `);
+        return undefined;
+      }
+      verArr[1] = Number(verArr[1]) + 1;
+      return verArr.join('.');
+    case 3: //Major
+      if (isBeta) {
+        return currentVersion.split('-')[0];
+      }
+      verArr[0] = Number(verArr[0]) + 1;
+      return verArr.join('.');
+    case 4: // Beta
+      if (isBeta) { //Existing beta
+        verArr[3] = Number(verArr[3]) + 1;
+        return verArr.join('.');
+      }
+      verArr[0] = Number(verArr[0]) + 1;
+      return verArr.join('.') + '-beta.0';
+    default:
+      logger.error(`ERROR: Invalid input: ${versionIncrement}`);
+      return undefined;
+  }
+};
+
+const updateChangelog = (newVersion) => {
+  logger.log(`Updating CHANGELOG.md with a new release entry - ${newVersion}`);
+  const filePath = path.resolve(__dirname, '../CHANGELOG.md');
+  let changeLog = fs.readFileSync(filePath).toString();
+  const latestEntryIndex = changeLog.indexOf('## [');
+  const newEntry = `## [${newVersion}] - ${new Date().toISOString().slice(0, 10)}
+    \n### Added
+    \n### Removed
+    \n### Changed
+    \n### Fixed
+    \n`;
+  changeLog = changeLog.substring(0, latestEntryIndex) + newEntry + changeLog.substring(latestEntryIndex);
+  fs.writeFileSync(filePath, changeLog);
+};
+
+const updateBaseBranch = (branchName) => {
+  logger.log(`Updating the base branch in .base-branch to ${branchName}`);
+  const filePath = path.resolve(__dirname, '../.base-branch');
+  fs.writeFileSync(filePath, `origin/${branchName}`);
+}
+
+const versionBump = async (option, branchName) => {
+  if (!option) {
+    logger.log('Choose one of the following options to bump the next version:');
+    logger.log('  1. Patch');
+    logger.log('  2. Minor');
+    logger.log('  3. Major');
+    logger.log('  4. Beta');
+    option = Number(await prompt(''));
+  }
+  const newVersion = getNewVersion(currentVersion, option);
+  if (!newVersion) {
+    quit(1);
```

Comment:
log?

Reply:
it is already log the error in getNewVersion.